lib/actions/posts.ts

```diff
@@ -1,19 +1,52 @@
 "use server";
 
-import { PostType } from "~/types";
-import { testPosts } from "~/constants";
+import { PostType, Note as TypeOfNote } from "~/types";
+
+import { NoteSchema } from "~/lib/validators/note.schema";
+import { connectToDB } from "~/utils/database";
+import { getSession } from "~/auth/auth";
+import Note from "~/models/note";
+import User from "~/models/user";
+import { NoteType } from "~/constants";
+
+export async function getAllUserTags() {
+  await connectToDB();
+  const session = await getSession();
+  const sessionUser = session?.user;
+
+  //get the posts of teh user from teh database
+  const allPosts = await Note.find({ creator: sessionUser?.id });
+
+  let allTags: string[] = [];
+  allPosts.forEach((post) => {
+    post.tags.forEach((tag: string) => {
+      if (!allTags.includes(tag)) {
+        allTags.push(tag);
+      }
+    });
+  });
```

Comment:
```suggestion
   const allTags = await Note.distinct("tags", { creator: sessionUser?.id });
```
The 'distinct' method in mongoose will give all the unique entries and returns it in an array. 

https://www.mongodb.com/docs/manual/reference/method/db.collection.distinct/